session/session.go

```diff
@@ -188,6 +191,27 @@ type session struct {
 	client kv.Client
 }
 
+var _ pd.TsoReqAlloc = &tsoReqAlloc{}
+
+type tsoReqAlloc struct {
+	pool []*pd.TsoRequest
+}
+
+func (s *tsoReqAlloc) Get() *pd.TsoRequest {
+	if len(s.pool) == 0 {
+		return pd.NewReqAlloc()
+	}
+	var req *pd.TsoRequest
+	req, s.pool = s.pool[len(s.pool)-1], s.pool[:len(s.pool)-1]
+	return req
+}
+
+func (s *tsoReqAlloc) Put(req *pd.TsoRequest) {
+	if len(s.pool) < 2 {
+		s.pool = append(s.pool, req)
```

Comment:
do we need to assure resued object is properly reset to intial memory state, or this has been done by the user?

Reply:
this code is useless and has be deleted, no we will use per session sync.Pool - - https://github.com/pingcap/pd/pull/1702#discussion_r317089597